core/tests/app/test_app.py

```diff
@@ -505,17 +561,174 @@ def test_set_empty_full_screen_window_list(event_loop):
     window1 = toga.Window()
     window2 = toga.Window()
 
-    assert not app.is_full_screen
+    with pytest.warns(
+        DeprecationWarning,
+        match=r"`App.is_full_screen` is deprecated. Use `App.is_presentation_mode` instead.",
+    ):
+        assert not app.is_full_screen
 
     # Change the screens that are full screen
-    app.set_full_screen(window1, window2)
-    assert app.is_full_screen
-    assert_action_performed_with(app, "enter_full_screen", windows=(window1, window2))
-
+    with pytest.warns(
+        DeprecationWarning,
+        match=r"`App.set_full_screen\(\)` is deprecated. Use `App.enter_presentation_mode\(\)` instead.",
+    ):
+        app.set_full_screen(window1, window2)
+    with pytest.warns(
+        DeprecationWarning,
+        match=r"`App.is_full_screen` is deprecated. Use `App.is_presentation_mode` instead.",
+    ):
+        assert app.is_full_screen
+    assert_action_performed_with(
+        app,
+        "enter presentation mode",
+        screen_window_dict={app.screens[0]: window1, app.screens[1]: window2},
+    )
     # Exit full screen mode by setting no windows full screen
-    app.set_full_screen()
-    assert not app.is_full_screen
-    assert_action_performed_with(app, "exit_full_screen", windows=(window1, window2))
+    with pytest.warns(
+        DeprecationWarning,
+        match=r"`App.set_full_screen\(\)` is deprecated. Use `App.enter_presentation_mode\(\)` instead.",
+    ):
+        app.set_full_screen()
+    with pytest.warns(
+        DeprecationWarning,
+        match=r"`App.is_full_screen` is deprecated. Use `App.is_presentation_mode` instead.",
+    ):
+        assert not app.is_full_screen
+    assert_action_performed(
+        app,
+        "exit presentation mode",
+    )
+
+
+def test_presentation_mode_with_windows_list(event_loop):
+    """The app can enter presentation mode with a windows list."""
+    app = toga.App(formal_name="Test App", app_id="org.example.test")
+    window1 = toga.Window()
+    window2 = toga.Window()
+
+    assert not app.is_presentation_mode
+
+    # Entering presentation mode with an empty windows list, is a no-op:
```

Comment:
Done.

Reply:
As previously mentioned in https://github.com/beeware/toga/pull/2473#discussion_r1642620415, I couldn't directly use the following in parametrize:
```
@pytest.mark.parametrize(
    "windows",
    [
        [],
        [toga.Window()],
        [toga.Window(), toga.Window()],
   ]
)
```
 
So, I used:
https://github.com/beeware/toga/blob/a276f22d0f0242f844f7a1c58b3f7ca8399943c2/core/tests/app/test_app.py#L468-L478

This works properly in python 3.10, but fails on python 3.12 with the following error:
```
=================================== ERRORS ====================================
___________________ ERROR collecting tests/app/test_app.py ____________________
tests\app\test_app.py:471: in <module>
    (toga.App(formal_name="Test App", app_id="org.example.test"), []),
..\.tox\py-cov\Lib\site-packages\toga\app.py:419: in __init__
    self._create_impl()
..\.tox\py-cov\Lib\site-packages\toga\app.py:422: in _create_impl
    self.factory.App(interface=self)
..\.tox\py-cov\Lib\site-packages\toga_dummy\app.py:24: in __init__
    self.loop = asyncio.get_event_loop()
C:\hostedtoolcache\windows\Python\3.12.4\x64\Lib\asyncio\events.py:697: in get_event_loop
    warnings.warn('There is no current event loop',
E   DeprecationWarning: There is no current event loop
=========================== short test summary info ===========================
ERROR tests/app/test_app.py - DeprecationWarning: There is no current event loop
!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
============================== 1 error in 4.86s ===============================
py-cov: exit 2 (6.45 seconds) D:\a\toga\toga\core> python -m coverage run -m pytest -vv --color yes pid=5480
  py-cov: FAIL code 2 (15.75=setup[1.41]+cmd[7.89,6.45] seconds)
  evaluation failed :( (15.92 seconds)
Error: Process completed with exit code 2.
```

Since, it gives a DeprecationWarning, is there a new way to parameterize this?

Reply:
You can't create the instances in the `parameterize()` because those values are constructed when the file is parsed. Instead, pass in a list of dictionaries, one for each instance you want to create, then create an instance for each provided dictionary.